packages/components/src/components/switch/switch.scss

```
@@ -0,0 +1,228 @@
+@use "@db-ui/foundations/build/scss/variables";
+@use "@db-ui/foundations/build/scss/colors";
+@use "@db-ui/foundations/build/scss/fonts";
+@use "@db-ui/foundations/build/scss/helpers";
+@use "@db-ui/foundations/build/scss/icons";
+@use "../../styles/form-components";
+
+$switch-fixed-padding: helpers.to-rem(2);
+$switch-border-workaround: calc(
+	#{variables.$db-border-height-2xs} / 2 * var(--device-pixel-ratio, 0)
+);
+
+$switch-active-thumb-size: calc(
+	#{form-components.$font-size-height} - #{$switch-border-workaround}
+);
+
+$checked-active-transition-size: calc(
+	#{form-components.$font-size-height} - #{$switch-fixed-padding} * 2
+);
+
+%active-transition {
+	&::before {
+		inline-size: calc(
+			#{$switch-active-thumb-size} - #{$switch-fixed-padding} * 2 + #{variables.$db-spacing-fixed-2xs}
+		);
+	}
+
+	&:checked {
+		&::before {
+			block-size: $checked-active-transition-size;
+		}
+	}
+}
+
+.db-switch {
+	@extend %check-element;
+	justify-content: space-between;
+	user-select: none;
+
+	@include helpers.active {
+		& > input {
+			@extend %active-transition;
+		}
+	}
+
+	& > input {
+		--db-icon-margin-start: 0;
+		--thumb-position: calc(0% - #{$switch-border-workaround});
+		@extend %db-overwrite-font-size-sm;
+
+		@include icons.set-icon("close", "after");
+		appearance: none;
+
+		inline-size: calc(
+			#{form-components.$font-size-height} * 2 + #{$switch-fixed-padding}
+		);
+		block-size: form-components.$font-size-height;
+		box-sizing: content-box;
+
+		display: flex;
+		align-items: center;
+		gap: $switch-fixed-padding;
+
+		background: colors.$db-current-color-bg-enabled;
```

```suggestion
		background-color: colors.$db-current-color-bg-enabled;
```

the variables value is always supposed to be a color, isn't it ?